examples/lang2/erc721/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(proc_macro_hygiene)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang2 as ink;
+use ink_core::{
+    memory::format,
+    storage,
+};
+
+#[ink::contract(version = "0.1.0")]
+mod erc721 {
+    #[ink(storage)]
+    struct Erc721 {
+        /// Mapping from token to owner.
+        token_owner: storage::HashMap<Hash, AccountId>,
+        /// Mapping from token to approvals users.
+        token_approvals: storage::HashMap<Hash, AccountId>,
+        /// Mapping from owner to number of owned token.
+        owned_tokens_count: storage::HashMap<AccountId, u128>,
+        /// Mapping from owner to operator approvals
+        operator_approvals: storage::HashMap<AccountId, storage::HashMap<AccountId, bool>>,
+        /// None Account to validate account
+        none_account: storage::Value<AccountId>,
+    }
+
+    /// Event emitted when a token transfer occurs
+    #[ink(event)]
+    struct Transfer {
+        #[ink(topic)]
+        from: Option<AccountId>,
+        #[ink(topic)]
+        to: AccountId,
+        #[ink(topic)]
+        id: Hash,
+    }
+
+    /// Event emited when a token approve occurs
+    #[ink(event)]
+    struct Approval {
+        #[ink(topic)]
+        from: AccountId,
+        #[ink(topic)]
+        to: AccountId,
+        #[ink(topic)]
+        id: Hash,
+    }
+
+    /// Event emitted when an operator is enabled or disabled for an owner.
+    /// The operator can manage all NFTs of the owner.
+    #[ink(event)]
+    struct ApprovalForAll {
+        #[ink(topic)]
+        owner: AccountId,
+        #[ink(topic)]
+        operator: AccountId,
+        approved: bool
+    }
+
+    impl Erc721 {
+        #[ink(constructor)]
+        fn new(&mut self) {
+            self.none_account.set(AccountId::from([0x0; 32]));
+        }
+
+        /// ===========================
+        /// NOT REQUIRED, JUST FOR TEST
+        /// ===========================
+        #[ink(message)]
+        fn mint(&mut self, id: Hash) -> bool {
+            let caller = self.env().caller();
+            let owner = self.owner_of_or_none(id);
+            if owner != *self.none_account {
+                self.env().println(&format!("Erc721::mint::token is exist"));
```

Comment:
Generally we don't want to put `println` into example contract codes.
Users should only make use of `println` for debugging purposes while developing.